skxray/speckle_visibility/speckle_visibility.py

```diff
@@ -0,0 +1,290 @@
+        to the pixel list
+        {ROI 1 : intensities of the pixels of ROI 1, ROI 2 : intensities of
+         the pixels of ROI 2}
+    """
+
+    if labels.shape != image.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+
+    #labels, indices = corr.extract_label_indices(label_array)
+    label_num = np.unique(labels)[1:]
+
+    #intensity_distribution = {}
+
+    #for n in label_num:
+    #    value = (np.ravel(image_array)[indices[labels == n].tolist()])
+    #    intensity_distribution[n] = value
+
+    return {n: image[labels == n] for n in range(1, np.max(labels))}
+
+
+def time_bin_edges(number=2, number_of_images=50):
+    """
+    This will provide the geometric series for the integration.
+    Last values of the series has to be less than or equal to number
+    of images
+    ex:
+        1, 2, 4, 8, 16, ...
+        1, 3, 9, 27, ...
+
+    Parameters
+    ----------
+    number : int, optional
+        time steps for the integration
+
+
+    number_of_images : int, 50
+        number of images
+
+    Return
+    ------
+    time_bin : list
+        time binning
+
+    Note
+    ----
+    :math ::
+     a + ar + ar^2 + ar^3 + ar^4 + ...
+
+     a - first term in the series
+     r - is the common ratio
+    """
+
+    time_bin = [1]
+
+    while time_bin[-1]*number < number_of_images:
+        time_bin.append(time_bin[-1]*number)
+    return time_bin
+
+
+def mean_intensity_sets(images_set, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for image sets
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int_labels : dict
+        average intensity of each ROI as a dictionary
+        {roi 1: average intensities, roi 2 : average intensities}
+
+    """
+    return {n+1 : mean_intensity(images_set[n],
+                                 labels) for n in range(len(images_set))}
+
+
+def mean_intensity(images, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int : array
+        mean intensity of each ROI for the set of images as an array
+        shape (number of images in the set, number of labels)
+
+    """
+    if labels.shape != images[0].shape[0:]:
+        raise ValueError("Shape of the images should be equal to"
+                         " shape of the label array")
+
+    index = np.unique(labels)[1: ]
```

Comment:
I think this should be `index = np.arange(1, np.max(labels) + 1)`.  If you have labels with gaps, you should probably use the relabel method to get the ROIs labeled sequentially again.